Mage.Sets/src/mage/cards/t/ThrastaTempestsRoar.java

```diff
@@ -0,0 +1,88 @@
+package mage.cards.t;
+
+        import java.util.UUID;
+        import mage.MageInt;
+        import mage.abilities.Ability;
+        import mage.abilities.common.SimpleStaticAbility;
+        import mage.abilities.dynamicvalue.DynamicValue;
+        import mage.abilities.effects.Effect;
+        import mage.abilities.effects.common.continuous.GainAbilitySourceEffect;
+        import mage.abilities.effects.common.cost.SpellCostReductionForEachSourceEffect;
+        import mage.abilities.hint.ValueHint;
+        import mage.abilities.keyword.HasteAbility;
+        import mage.abilities.keyword.HexproofAbility;
+        import mage.abilities.keyword.TrampleAbility;
+        import mage.abilities.keyword.TrampleOverPlaneswalkersAbility;
+        import mage.cards.CardImpl;
+        import mage.cards.CardSetInfo;
+        import mage.constants.*;
+        import mage.game.Game;
+        import mage.watchers.common.CastSpellLastTurnWatcher;
+
+/**
+ * @author Fubs
+ */
+public final class ThrastaTempestsRoar extends CardImpl {
+
+    public ThrastaTempestsRoar(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.CREATURE},"{10}{G}{G}");
+        addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(7);
+        this.toughness = new MageInt(7);
+
+        //"This spell costs 3 less to cast for each other spell cast this turn"
+        ThrastaDynamicValue spellCastCount = new ThrastaDynamicValue();
```

Comment:
I tested using OtherSpellsCastThisTurnCount, but it made the casting cost 3 more than the correct value. The current version works correctly. I guess the "-1" logic is not needed for this card because the spells cast count does not increment before the cost is paid.

I could also just extract OtherSpellsCastThisTurnCount, and add +1 before it is used. Would that be better?

Reply:
> and add +1 before it is used. Would that be better?

Hmm. Nope, it's not good. So if it works without "-1" then keep your code as is -- just add comment like "no needs -1 cause cast count increases after current spell cast/reduction".